src/feat/wave-reader.h

```diff
@@ -171,6 +172,11 @@ class WaveHolder {
     t_.Swap(&(other->t_));
   }
 
+  bool ExtractRange(const WaveHolder &other, const std::string &range) {
+    KALDI_ERR << "ExtractRange is not defined for this type of holder.";
```

Comment:
It seems that all holders except one or two need an exact clone of this function. I would rather use some SFINAE magic (like that in http://stackoverflow.com/a/264088/1149924) at call site, to detect if given holder type has the member. I can draft a patch for that if you think it would be less ugly than this.